lib/index.ts

```diff
@@ -600,12 +600,14 @@ export async function exec(argv2: string[]) {
   if (configJson) {
     marker = {
       config: configJson,
+      toplevelConfig: configJson,
```

Comment:
About this, I think it could make sense to create a `config.ts` file like:

```ts
class Config {
  private static instance: Config;
  private config: any;

  private constructor() {
    this.config = {};
  }

  public static getInstance(): Config {
    if (!Config.instance) {
      Config.instance = new Config();
    }

    return Config.instance;
  }

  public set(config: any): void {
    this.config = config;
  }

  public get(): any {
    return this.config;
  }
}

export default Config;
```

then use it code-wise like:

```ts
import Config from './config';

const config = Config.getInstance();

// Set the parsed config on main
config.set(parsedArgs);

// Get the parsed arguments when needed
const args = config.get();
```

This is a cleaner approach IMO and would simplify code a lot

Reply:
The singleton way is cleaner IMO